### PR TITLE
Bring skeleton

### PR DESCRIPTION
**Pull request by aprudyus**
This PR adds the `ApplicationContext` to the app with the support of basic annotations like `@Bean` and `@Autowired`. `ApplicationContext` tests are also added.

### CODE REVIEW

File: src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            postProcess();
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new ApplicationContextInitializationException(
+                format("ApplicationContext initialization has failed: %s", e.getMessage())
+            );
+        }
+    }
+
+    private void initBeans(Set<Class<?>> beanClasses) throws NoSuchMethodException, InvocationTargetException,
```
Comment (Tarasik91) [inline src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java]: I would handle NoSuchMethodException, InvocationTargetException inside a method and throw exception with error message 'Default constructor wasn't find for #className '

File: src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    }
+
+    private String resolveBeanName(Class<?> type) {
+        String beanName = type.getAnnotation(Bean.class).value();
+        return isNotBlank(beanName) ? beanName : resolveBeanName(type.getSimpleName());
+    }
+
+    private String resolveBeanName(String typeName) {
+        return typeName.substring(0, 1).toLowerCase() + typeName.substring(1);
```
Comment (Tarasik91) [inline src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java:81]: Could we create a class with a name consisting of  one letter?

**Review by Tarasik91 — COMMENTED** [review]
Smt wrong with test coverage step of sonarcude

File: src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        Reflections beanScanner = new Reflections((Object[]) basePackages);
+        Set<Class<?>> beanClasses = beanScanner.getTypesAnnotatedWith(Bean.class, true);
+
+        if (beanClasses.isEmpty()) {
+            return;
+        }
+
+        try {
+            initBeans(beanClasses);
+            postProcess();
```
Comment (Tarasik91) [inline src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java:38]: Does make sense move the logic of postProcess to separately class?

File: src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java (hunk 1 above)
Comment (aprudyus) [reply src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java]: Done

File: src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java (hunk 2 above)
Comment (aprudyus) [reply src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java:81]: Yes, I added a few test cases to cover that.

File: src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java (hunk 3 above)
Comment (aprudyus) [reply src/main/java/com/hoverla/bring/context/ApplicationContextImpl.java:38]: It does, but let's do it in scope of this task https://github.com/users/overpathz/projects/1?pane=issue&itemId=19372572

**Review by Tarasik91 — APPROVED** [review]

**Pull request merged**